modules/tensor_mechanics/include/auxkernels/RankTwoScalarAux.h

```diff
@@ -29,12 +29,18 @@ class RankTwoScalarAux : public AuxKernel
 
 protected:
   virtual Real computeValue();
-  Real calcEigenValues();
 
   const MaterialProperty<RankTwoTensor> & _tensor;
   MooseEnum _scalar_type;
 
+  /// whether or not selected_qp has been set
+  const bool _has_selected_qp;
+
+  /// The std::vector will be evaluated at this quadpoint only
+  const unsigned int _selected_qp;
+
 private:
```

Comment:
I guess I don't see why this has to be `private`. Do we really need to prevent the use of this function by potential derived classes? What are we protecting here?